framework/src/base/FEProblem.C

```diff
@@ -416,7 +416,7 @@ void FEProblem::initialSetup()
     Moose::setup_perf_log.push("initial adaptivity", "Setup");
     unsigned int n = adaptivity().getInitialSteps();
 
-    if (n && !_app.isUltimateMaster())
+    if (n && !_app.isUltimateMaster() && _app.isRestarting())
```

Comment:
Is this a bugfix?  This should probably be a separate PR.

BTW: I might be removing this restriction altogether with the MultiApp restart work I currently have going...